src/oop/object_interactions/medium/q6/MainTest.java

```diff
@@ -0,0 +1,191 @@
+            _assertEquals(seat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(pass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            int updateSeat = (int) test[num][2];
+            int updatePass = (int) test[num][3];
+            Object output = testClass.callMethod(getMaxSeatsMethodName, classInstance);
+            _assertEquals(seat, output,
+                    "You have modified the provided " + getMaxSeatsMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(getNumPassengersMethodName, classInstance);
+            _assertEquals(pass, output,
+                    "You have modified the provided " + getNumPassengersMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            testClass.callMethod(setMaxSeatsMethodName, new Object[][]{{updateSeat, int.class}}, classInstance);
+            _assertEquals(updateSeat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "You have modified the provided " + setMaxSeatsMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            testClass.callMethod(setNumPassengersMethodName, new Object[][]{{updatePass, int.class}}, classInstance);
+            _assertEquals(updatePass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided " + setNumPassengersMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(20, 10, 40, 35),
+                Arguments.of(10, 20, 15, 30),
+                Arguments.of(150, 200, 100, 50),
+                Arguments.of(20, 20, 10, 20)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctTransferPassengersMethod(int seat, int pass, int busySeat, int busyPass) throws Throwable {
+        Object[][] arguments = {
+                {seat, int.class},
+                {pass, int.class}
+        };
+        Object[][] busyArguments = {
+                {busySeat, int.class},
+                {busyPass, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object busyInstance = testClass.createInstance(busyArguments);
+        Object[][] methodArguments = {
+                {busyInstance, Airplane.class}
+        };
+        Object output = null;
```

Comment:
use .hasMethod before the if statement. It gets rid of the need for an output variable and the assertNull